includes/admin/class-sensei-wccom-connect-notice.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * File containing the class Sensei_WCCOM_Connect_Notice
+ *
+ * @package sensei-lms
+ * @since   3.11.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class that handles showing the WooCommerce.com connect notice.
+ *
+ * @access private
+ *
+ * @class Sensei_WCCOM_Connect_Notice
+ */
+class Sensei_WCCOM_Connect_Notice {
+	const DISMISS_NOTICE_NONCE_ACTION   = 'sensei-lms-cancelled-wccom-connect-dismiss';
+	const DISMISSED_NOTIFICATION_OPTION = 'sensei-cancelled-wccom-connect-dismissed';
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		add_filter( 'admin_notices', [ $this, 'add_admin_notice' ] );
+		add_action( 'wp_ajax_sensei_dismiss_wccom_connect_notice', [ $this, 'handle_notice_dismiss' ] );
+	}
+
+	/**
+	 * Check if WooCommerce.com connection has been made.
+	 *
+	 * @return bool
+	 */
+	private function is_wccom_connected() {
+		if ( ! class_exists( 'WC_Helper_Options' ) ) {
+			// We can't do anything until WooCommerce is activated. There is already a notice for that.
+			return true;
+		}
+
+		$auth = WC_Helper_Options::get( 'auth' );
+
+		return ! empty( $auth['access_token'] );
+	}
+
+	/**
+	 * Get the connect URL.
+	 */
+	private function get_wccom_connect_url() {
+		return add_query_arg(
+			[
+				'page'              => 'wc-addons',
+				'section'           => 'helper',
+				'wc-helper-connect' => 1,
+				'wc-helper-nonce'   => wp_create_nonce( 'connect' ),
+			],
+			admin_url( 'admin.php' )
+		);
+	}
+
+	/**
+	 * Output the admin notice.
+	 *
+	 * @access private
+	 */
+	public function add_admin_notice() {
+		if (
+			$this->is_wccom_connected()
+			|| $this->is_notice_dismissed()
+			|| ! $this->has_wccom_sensei_extension()
+			|| ! $this->can_see_notice_on_screen()
+			|| ! $this->can_update_plugins()
+		) {
+			return;
+		}
+
+		wp_enqueue_script( 'sensei-dismiss-notices' );
+
+		$wccom_connect_url = $this->get_wccom_connect_url();
+		?>
+		<div id="sensei-lms-wccom-connect-notice" class="notice sensei-notice is-dismissible" data-dismiss-action="sensei_dismiss_wccom_connect_notice"
+				data-dismiss-nonce="<?php echo esc_attr( wp_create_nonce( self::DISMISS_NOTICE_NONCE_ACTION ) ); ?>">
+			<p>
+				<a href="<?php echo esc_url( $wccom_connect_url ); ?>" class="button button-primary">
+					<?php esc_html_e( 'Connect account', 'sensei-lms' ); ?>
+				</a>
+				<?php
+				esc_html_e(
+					'Get notified about new features and updates by connecting your WooCommerce account.',
```

Comment:
One small tweak...I think this should say WooCommerce.com account to reduce confusion.

Reply:
Updated the message 4dc22d45fef13d83917ec623b274acbcb2b44937